Goal of collecting dataset:
--------------
Our objective is to leverage both classification and clustering techniques to develop a machine-learning model that predicts the risk of heart attacks in individuals. For classification, we aim to train our model to categorize individuals who have heart attack risks from those who don’t based on a comprehensive set of health and lifestyle factors. As for clustering, our model will be able to explore the underlying patterns within the data, identify distinct groups of individuals who share similar risk profiles, and create a set of clusters for these individuals, which in return will be used to predict new individuals’ results. 

Sourse of dataset:
----
Heart Attack Prediction: 
https://www.kaggle.com/datasets/m1relly/heart-attack-prediction/data

Read dataset:
--

In [1]:
import pandas as pd
df=pd.read_csv('Dataset/heart.attack.csv')

General information about the dataset:
--
- Number of attributes: 12
- Number of objects: 4000
- Class label: Heart Attack Risk

In [2]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4000 entries, 0 to 3999
Data columns (total 12 columns):
 #   Column             Non-Null Count  Dtype 
---  ------             --------------  ----- 
 0   Patient ID         4000 non-null   object
 1   Age                4000 non-null   int64 
 2   Sex                4000 non-null   object
 3    Cholesterol       4000 non-null   int64 
 4   Blood Pressure     4000 non-null   object
 5    Heart Rate        4000 non-null   int64 
 6   Diabetes           4000 non-null   int64 
 7   Family History     4000 non-null   int64 
 8   Smoking            4000 non-null   int64 
 9   Diet               4000 non-null   object
 10  Continent          4000 non-null   object
 11  Heart Attack Risk  4000 non-null   int64 
dtypes: int64(7), object(5)
memory usage: 375.1+ KB


As we can see, there are 4,000 objects (numbered from 0 to 3,999), a total of 12 columns, and all objects are non-null. Regarding data types, there are 7 integer attributes and 5 object attributes.